sbi/inference/posteriors/base_posterior.py

```diff
@@ -575,9 +575,21 @@ def __str__(self):
             else ""
         )
 
-        # The net might be sequential because it can have a standardization net. Hence,
-        # we only access its last entry if it is a nn.Sequential.
-        actual_net = self.net[-1] if isinstance(self.net, nn.Sequential) else self.net
+        # For SNRE the net is sequential because it has a standardization net. Hence,
+        # we only access its last entry.
+        try:
+            first_net = self.net[0]
+            # Why not `isinstance(first_net, StandardizeInputs)`? Because
+            # `StandardizeInputs` is defined within a function in
+            # neural_nets/classifier.py and can not be imported here.
+            if first_net.__class__.__name__ == "StandardizeInputs":
```

Comment:
I think it's completely fine to write self.net[0] - it's absolutely clear, and there's no reuse of the name...